crates/pop-wallet-integration/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+use axum::{
+	routing::{get, post},
+	Router,
+};
+use serde::Serialize;
+use std::{path::PathBuf, sync::Arc};
+use tokio::sync::{oneshot, Mutex};
+use tower_http::services::ServeDir;
+
+/// Data to be sent to frontend for signing.
+#[derive(Serialize, Debug)]
+pub struct Data {
+	chain_rpc: String,
+	data_type: DataType,
+}
+
+/// The type of transaction with specific data for signing (contract or parachain).
+#[derive(Serialize, Debug, PartialEq)]
+pub enum DataType {
+	/// Parachain call, where Vec<u8> is the encoded call data.
+	Parachain(Vec<u8>),
+	Contract(ContractArgs),
+}
+
+/// Contract specific data variations.
+pub mod contracts {
+	use super::Serialize;
+	#[derive(Serialize, Debug, PartialEq)]
+	pub struct ContractArgs {
+		call_type: ContractCallType,
+		storage_deposit_limit: Option<String>,
+		/// The binary (wasm, polkavm) of the contract.
+		code: Option<Vec<u8>>,
+	}
+
+	#[derive(Serialize, Debug, PartialEq)]
+	pub enum ContractCallType {
+		// no unique fields.
+		Upload,
+		Instantiate(InstantiateArgs),
+		Call(CallArgs),
+	}
+
+	/// Arguments for instantiating a contract.
+	#[derive(Serialize, Debug, PartialEq)]
+	pub struct InstantiateArgs {
+		constructor: String,
+		args: Vec<String>,
+		value: String,
+		gas_limit: Option<u64>,
+		proof_size: Option<u64>,
+		salt: Option<Vec<u8>>,
+	}
+
+	/// Arguments for calling a contract.
+	#[derive(Serialize, Debug, PartialEq)]
+	pub struct CallArgs {
```

Comment:
For the data we pass to contracts, can we construct the `call_data` in `pop-cli` in the same way as we do for parachain extrinsics. This is a feature we didn’t have before, but for `pop call parachain` we are constructing the extrinsic (see https://github.com/r0gue-io/pop-cli/blob/feat-call-ui-extrinsics-limited/crates/pop-parachains/src/call/mod.rs#L30). Before open the UI we have all the fields required to construct the Contracts(or Revive)::call  extrinsic so we can get the `call_data` and have the same behaviour as Parachain extrinsics?

With this approach, we will still need to modify how the `dry_run` is performed in the `pop-cli`

Reply:
https://github.com/r0gue-io/pop-cli/pull/362/commits/cfe41f00110093a32bd2006efafad443b11d0085